app/src/main/res/layout/fragment_add_login.xml

```diff
@@ -111,7 +112,7 @@
         android:colorControlActivated="?attr/textPrimary"
         android:textColor="?attr/textPrimary"
         android:contentDescription="@string/saved_login_username_description"
-        app:layout_constraintEnd_toEndOf="@id/usernameHeader"
+        app:layout_constraintEnd_toStartOf="@id/usernameHeader"
```

Comment:
Fix looks good 👍 

~~I don't think this should have changed? This is still displaying ok is because of the set width which should probably use 0dp as the constraints look set to anchor with usernameHeader~~ Ah actually, I understand this is consistent with inputLayoutPassword.   

This may be out of scope for this change (interested to get your thoughts) but does it seems odd to have:
```
android:layout_width="match_parent"
...
app:layout_constraintEnd_toStartOf="@id/usernameHeader"
app:layout_constraintStart_toStartOf="@id/usernameHeader"
```

Using constraint layout, I would have expected something more like:
```
android:layout_width="0dp"
...
app:layout_constraintEnd_toEndOf="parent"
app:layout_constraintStart_toStartOf="parent"
```

Or have I missed something altogether?

Reply:
I appreciate that some of this may be outside the scope of your issue. I wonder if we also create new issue to tidy up this file a bit? For instance the ‘clear’ button for user has different behaviour to ‘site’ & ‘password’ which may/may not be wanted

Reply:
I added some refactors to the xml layout, that are already on the second commit of this PR. Thanks for pointing out those incoherences!

I'll also file a new issue to report the differences of behaviours between the `clear` buttons of this field.

Reply:
I've juste filed a bug for this issue:
https://bugzilla.mozilla.org/show_bug.cgi?id=1816148